autoware_utils/include/autoware_utils/ros/transform_listener.hpp

```diff
@@ -15,73 +15,13 @@
 #ifndef AUTOWARE_UTILS__ROS__TRANSFORM_LISTENER_HPP_
 #define AUTOWARE_UTILS__ROS__TRANSFORM_LISTENER_HPP_
 
-#include <rclcpp/rclcpp.hpp>
-
-#include <geometry_msgs/msg/transform_stamped.hpp>
-
-#include <tf2_ros/buffer.h>
-#include <tf2_ros/create_timer_ros.h>
-#include <tf2_ros/transform_listener.h>
-
-#include <memory>
-#include <string>
+#include <autoware_utils_tf2/transform_listener.hpp>
 
 namespace autoware_utils
 {
-class TransformListener
-{
-public:
-  explicit TransformListener(rclcpp::Node * node)
-  : clock_(node->get_clock()), logger_(node->get_logger())
-  {
-    tf_buffer_ = std::make_shared<tf2_ros::Buffer>(clock_);
-    auto timer_interface = std::make_shared<tf2_ros::CreateTimerROS>(
-      node->get_node_base_interface(), node->get_node_timers_interface());
-    tf_buffer_->setCreateTimerInterface(timer_interface);
-    tf_listener_ = std::make_shared<tf2_ros::TransformListener>(*tf_buffer_);
-  }
-
-  geometry_msgs::msg::TransformStamped::ConstSharedPtr get_latest_transform(
-    const std::string & from, const std::string & to)
-  {
-    geometry_msgs::msg::TransformStamped tf;
-    try {
-      tf = tf_buffer_->lookupTransform(from, to, tf2::TimePointZero);
-    } catch (tf2::TransformException & ex) {
-      RCLCPP_WARN_THROTTLE(
-        logger_, *clock_, 5000, "failed to get transform from %s to %s: %s", from.c_str(),
-        to.c_str(), ex.what());
-      return {};
-    }
-
-    return std::make_shared<const geometry_msgs::msg::TransformStamped>(tf);
-  }
-
-  geometry_msgs::msg::TransformStamped::ConstSharedPtr get_transform(
-    const std::string & from, const std::string & to, const rclcpp::Time & time,
-    const rclcpp::Duration & duration)
-  {
-    geometry_msgs::msg::TransformStamped tf;
-    try {
-      tf = tf_buffer_->lookupTransform(from, to, time, duration);
-    } catch (tf2::TransformException & ex) {
-      RCLCPP_WARN_THROTTLE(
-        logger_, *clock_, 5000, "failed to get transform from %s to %s: %s", from.c_str(),
-        to.c_str(), ex.what());
-      return {};
-    }
-
-    return std::make_shared<const geometry_msgs::msg::TransformStamped>(tf);
-  }
 
-  rclcpp::Logger get_logger() { return logger_; }
+using namespace autoware_utils_tf2;  // NOLINT(build/namespaces)
```

Comment:
We might want to have a similar file for autoware_util_pcl for backward compatibility.

Reply:
I added the `autoware_utils_pcl` headers for compatibility. https://github.com/autowarefoundation/autoware_utils/pull/29/commits/8a83320f9243d58b5be38182e58c4bd4186d1c56